DocumentVectorPipelineFunctions/BlobTriggerFunction.cs

```diff
@@ -42,37 +44,49 @@ public async Task Run([BlobTrigger("documents/{name}", Connection = "AzureBlobSt
     private async Task HandleBlobCreateEventAsync(BlobClient blobClient)
     {
         this._logger.LogInformation("Analyzing document using DocumentAnalyzerService from blobUri: '{blobUri}' using layout: {layout}", blobClient.Name, "prebuilt-read");
-        var operation = await documentAnalysisClient.AnalyzeDocumentFromUriAsync(
-            WaitUntil.Completed,
-            "prebuilt-read",
-            blobClient.Uri);
-
-        var result = operation.Value;
-
-        this._logger.LogInformation("Extracted content from '{name}', # pages {pageCount}", blobClient.Name, result.Pages.Count);
 
         var cosmosDBClientWrapper = await CosmosDBClientWrapper.CreateInstance(cosmosClient, this._logger);
 
-        int totalChunksCount = 0;
-        var batchChunkTexts = new List<TextChunk>();
-        foreach (var chunk in TextChunker.FixedSizeChunking(result))
+        using (var blobStream = await blobClient.OpenReadAsync())
         {
-            batchChunkTexts.Add(chunk);
-            totalChunksCount++;
-
-            if (batchChunkTexts.Count >= MaxBatchSize)
+            byte[] buffer = new byte[BlobTriggerFunction.BufferSize];
+            int bytesRead;
+            while ((bytesRead = await blobStream.ReadAsync(buffer, 0, BlobTriggerFunction.BufferSize)) > 0)
             {
-                await this.ProcessCurrentBatchAsync(blobClient, cosmosDBClientWrapper, batchChunkTexts);
+                // Process the chunk with Document Intelligence
+                using (var memoryStream = new MemoryStream(buffer, 0, bytesRead))
+                {
+                    var operation = await documentAnalysisClient.AnalyzeDocumentAsync(
+                        WaitUntil.Completed,
+                        "prebuilt-read",
+                        memoryStream);
```

Comment:
yeah, simple text it can (i validated) but yes, it can't do complex document parsing. But I didn't want to do entire document because if it's too large and we currently don't have any limits put in application.

Do you think it is okay for now to read entire document and not worry about big size/memory allocations?

Reply:
Agreed - I think it's okay to read the entire document for now. This is in the customer's subscription, on their own documents anyway.

Reply:
sounds good!